plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/ViewportPluginImpl.kt

```diff
@@ -0,0 +1,389 @@
+import com.mapbox.maps.plugin.viewport.state.ViewportCameraState.TransitionToOverview
+import com.mapbox.maps.plugin.viewport.state.ViewportCameraStateChangedObserver
+import com.mapbox.maps.plugin.viewport.transition.MapboxViewportCameraStateTransition
+import com.mapbox.maps.plugin.viewport.transition.TransitionEndListener
+import com.mapbox.maps.plugin.viewport.transition.ViewportCameraStateTransition
+import com.mapbox.maps.plugin.viewport.transition.ViewportCameraTransition
+import com.mapbox.maps.plugin.viewport.transition.ViewportCameraTransitionOptions
+import java.util.concurrent.CopyOnWriteArraySet
+
+/**
+ * Mapbox default implementation for Viewport plugin.
+ */
+class ViewportPluginImpl : ViewportPlugin {
+  private var frameTransitionOptions = DEFAULT_FRAME_TRANSITION_OPT
+  private var runningAnimation: AnimatorSet? = null
+  private val transitionEndListeners = CopyOnWriteArraySet<TransitionEndListener>()
+  private val viewportCameraStateChangedObservers =
+    CopyOnWriteArraySet<ViewportCameraStateChangedObserver>()
+  private lateinit var cameraPlugin: CameraAnimationsPlugin
+  private lateinit var locationComponentPlugin: LocationComponentPlugin
+  private lateinit var mapDelegateProvider: MapDelegateProvider
+  private lateinit var stateTransition: ViewportCameraStateTransition
+
+  /**
+   * Describes an object that provides desired camera positions to [ViewportPlugin].
+   */
+  override lateinit var viewportDataSource: ViewportDataSource
+
+  /**
+   * Returns current [ViewportCameraTransition].
+   * @see registerViewportCameraStateChangedObserver
+   */
+  override var state: ViewportCameraState = Idle
+    private set(value) {
+      if (value != field) {
+        field = value
+        viewportCameraStateChangedObservers.forEach {
+          it.onViewportCameraStateChanged(value)
+        }
+      }
+    }
+
+  private val sourceUpdateObserver = ViewportDataSourceUpdateObserver(::updateFrame)
+
+  /**
+   * Executes a transition to [Following] state. When started, goes to [TransitionToFollowing]
+   * and to the final [Following] when ended. If transition is canceled, state goes to [Idle].
+   *
+   * The target camera position is obtained with [ViewportDataSource.getViewportData].
+   *
+   * @param stateTransitionOptionsBlock options that impact the transition animation from
+   * the current state to the requested state.
+   * Defaults to [ViewportCameraTransitionOptions.maxDurationMs] equal to 3500 millis.
+   * @param frameTransitionOptionsBlock options that impact the transition animations between
+   * viewport frames in the selected state.
+   * This refers to camera transition on each [ViewportDataSource] update when [Following] is engaged.
+   * Defaults to [ViewportCameraTransitionOptions.maxDurationMs] equal to 1000 millis.
+   * @param transitionEndListener invoked when transition ends.
+   */
+  @JvmOverloads
+  override fun requestNavigationCameraToFollowing(
```

Comment:
`requestNavigationCameraToFollowing` this wording is too navigation specific, would love to be able to generalize that

Reply:
same for all other overrides in this class, see if we can drop navigation naming

Reply:
yes, this PR is meant to be the initial PR to drop in as much as navigation logic and iterate the public API in follow up PRs, but will rename these interfaces in this PR.

Reply:
More class name refactoring is done in https://github.com/mapbox/mapbox-maps-android/pull/969